robot/impl/local_robot.go

```diff
@@ -776,14 +778,14 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	}
 }
 
-// CheckMaxInstance checks to see if the local robot has reached the maximum number of a specific service type.
-func (r *localRobot) CheckMaxInstance(f *registry.Service, name resource.Name) error {
+// checkMaxInstance checks to see if the local robot has reached the maximum number of a specific service type that are local.
+func (r *localRobot) checkMaxInstance(subtype resource.Subtype, max int) error {
 	maxInstance := 0
 	for _, n := range r.ResourceNames() {
-		if n.Subtype == name.Subtype {
+		if n.Subtype == subtype && !n.ContainsRemoteNames() {
 			maxInstance++
-			if maxInstance == f.MaxInstance {
-				return errors.Errorf("Max instance number reached for service type: %s", name.Subtype)
+			if maxInstance == max {
```

Comment:
I would have thought we'd want to return err only if `maxInstance > max`.

Reply:
in this case, we're checking the number of existing instances when creating a new instance - so equal means we should err